mlx provider doesn't compile anymore on master (cddcb137)

Compiling libfabric with the mlx provider turned on throws the following compilation errors:

```
  CC       prov/hook/src/src_libfabric_la-hook_eq.lo
  CC       prov/hook/src/src_libfabric_la-hook_wait.lo
  CC       prov/hook/src/src_libfabric_la-hook_xfer.lo
  CC       src/src_libfabric_la-rbtree.lo
  CC       src/src_libfabric_la-tree.lo
  CC       src/src_libfabric_la-fasthash.lo
  CC       prov/mlx/src/src_libfabric_la-mlx_tagged.lo
  CC       prov/mlx/src/src_libfabric_la-mlx_callbacks.lo
  CC       util/strerror.o
prov/mlx/src/mlx_callbacks.c: In function 'mlx_send_callback':
prov/mlx/src/mlx_callbacks.c:65:26: error: invalid application of 'sizeof' to incomplete type 'struct util_cq_err_entry'
   err = calloc(1, sizeof(struct util_cq_err_entry));
                          ^~~~~~
prov/mlx/src/mlx_tagged.c: In function 'mlx_tagged_recvmsg':
prov/mlx/src/mlx_callbacks.c:72:6: error: dereferencing pointer to incomplete type 'struct util_cq_err_entry'
   err->err_entry = (mlx_req->completion.error);
      ^~
prov/mlx/src/mlx_callbacks.c:76:42: error: 'struct util_cq' has no member named 'err_list'; did you mean 'ep_list'?
   slist_insert_tail(&err->list_entry, &cq->err_list);
                                          ^~
prov/mlx/src/mlx_tagged.c:105:26: error: invalid application of 'sizeof' to incomplete type 'struct util_cq_err_entry'
   err = calloc(1, sizeof(struct util_cq_err_entry));
                          ^~~~~~
prov/mlx/src/mlx_tagged.c:112:6: error: dereferencing pointer to incomplete type 'struct util_cq_err_entry'
   err->err_entry = (req->completion.error);
      ^~
prov/mlx/src/mlx_callbacks.c: In function 'mlx_recv_callback':
prov/mlx/src/mlx_callbacks.c:139:27: error: invalid application of 'sizeof' to incomplete type 'struct util_cq_err_entry'
    err = calloc(1, sizeof(struct util_cq_err_entry));
                           ^~~~~~
prov/mlx/src/mlx_tagged.c:113:42: error: 'struct util_cq' has no member named 'err_list'; did you mean 'ep_list'?
   slist_insert_tail(&err->list_entry, &cq->err_list);
                                          ^~
prov/mlx/src/mlx_callbacks.c:147:7: error: dereferencing pointer to incomplete type 'struct util_cq_err_entry'
    err->err_entry = (mlx_req->completion.error);
       ^~
prov/mlx/src/mlx_callbacks.c:148:43: error: 'struct util_cq' has no member named 'err_list'; did you mean 'ep_list'?
    slist_insert_tail(&err->list_entry, &cq->err_list);
```

I'm not particularly interested in this provider (I've enabled this provider by mistake in my local installation), but maybe someone should probably have a look.

--- a/prov/mlx/src/mlx_callbacks.c
+++ b/prov/mlx/src/mlx_callbacks.c
@@ -45,7 +45,7 @@ void mlx_send_callback(void *request,
 	struct util_cq *cq;
 	struct mlx_request *mlx_req = request;
 	struct fi_cq_tagged_entry *t_entry;
-	struct util_cq_err_entry *err;
+	struct util_cq_oflow_err_entry *err;
 
 	cq = mlx_req->cq;
 
@@ -62,18 +62,18 @@ void mlx_send_callback(void *request,
 
 	if (status != UCS_OK){
 		t_entry->flags |= UTIL_FLAG_ERROR;
-		err = calloc(1, sizeof(struct util_cq_err_entry));
+		err = calloc(1, sizeof(struct util_cq_oflow_err_entry));
 		if (!err) {
 			FI_WARN(&mlx_prov, FI_LOG_CQ,
 				"out of memory, cannot report CQ error\n");
 			goto fn;
 		}
 
-		err->err_entry = (mlx_req->completion.error);
-		err->err_entry.prov_errno = (int)status;
-		err->err_entry.err = MLX_TRANSLATE_ERRCODE(status);
-		err->err_entry.olen = 0;
-		slist_insert_tail(&err->list_entry, &cq->err_list);
+		err->comp = (mlx_req->completion.error);
+		err->comp.prov_errno = (int)status;
+		err->comp.err = MLX_TRANSLATE_ERRCODE(status);
+		err->comp.olen = 0;
+		slist_insert_tail(&err->list_entry, &cq->oflow_err_list);
 	}
 fn:
 	mlx_req->type = MLX_FI_REQ_UNINITIALIZED;
@@ -133,10 +133,10 @@ void mlx_recv_callback(void *request,
 		*t_entry = (mlx_req->completion.tagged);
 
 		if (status != UCS_OK) {
-			struct util_cq_err_entry* err;
+			struct util_cq_oflow_err_entry *err;
 			t_entry->flags |= UTIL_FLAG_ERROR;
 
-			err = calloc(1, sizeof(struct util_cq_err_entry));
+			err = calloc(1, sizeof(struct util_cq_oflow_err_entry));
 			if (!err) {
 				FI_WARN(&mlx_prov, FI_LOG_CQ,
 					"out of memory, cannot report CQ error\n");
@@ -144,8 +144,8 @@ void mlx_recv_callback(void *request,
 				goto fn;
 			}
 
-			err->err_entry = (mlx_req->completion.error);
-			slist_insert_tail(&err->list_entry, &cq->err_list);
+			err->comp = (mlx_req->completion.error);
+			slist_insert_tail(&err->list_entry, &cq->oflow_err_list);
 		}
 
 		if (cq->src){

--- a/prov/mlx/src/mlx_tagged.c
+++ b/prov/mlx/src/mlx_tagged.c
@@ -98,19 +98,19 @@ static ssize_t mlx_tagged_recvmsg(
 	*t_entry = (req->completion.tagged);
 
 	if (req->type == MLX_FI_REQ_UNEXPECTED_ERR) {
-		struct util_cq_err_entry* err;
+		struct util_cq_oflow_err_entry* err;
 		req->completion.error.olen -= req->completion.tagged.len;
 		t_entry->flags |= UTIL_FLAG_ERROR;
 
-		err = calloc(1, sizeof(struct util_cq_err_entry));
+		err = calloc(1, sizeof(struct util_cq_oflow_err_entry));
 		if (!err) {
 			FI_WARN(&mlx_prov, FI_LOG_CQ,
 				"out of memory, cannot report CQ error\n");
 			fastlock_release(&cq->cq_lock);
 			return -FI_ENOMEM;
 		}
-		err->err_entry = (req->completion.error);
-		slist_insert_tail(&err->list_entry, &cq->err_list);
+		err->comp = (req->completion.error);
+		slist_insert_tail(&err->list_entry, &cq->oflow_err_list);
 	}
 
 	ofi_cirque_commit(cq->cirq);

